### ISSUE TITLE
Nav offcanvas - creating a menu from the block list view transfers focus back to canvas

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#47018]
As reported in https://github.com/WordPress/gutenberg/issues/46939#issuecomment-1374652632, when inserting blocks _focus_ (note this is different to block _selection_) may be transferred to the canvas.

Instead focus should remain within/return to the context of the offcanvas from which you added the block.

**Comment by talldan** [comment]
It's weird because the `Inserter` component has `selectBlockOnInsert={ false }` specified:
https://github.com/WordPress/gutenberg/blob/a4879a68914e63b1cb9a51f6d18680463ac2b9f2/packages/block-editor/src/components/off-canvas-editor/appender.js#L36-L45

I guess that prop doesn't work.

Edit: I think the issue in the video is that a menu was created, and I think it's that causing the focus loss. In general usage I don't see any issues with focus being transferred.

**Comment by getdave** [comment]
I have a sneaking suspicion that "focus" and "selected" are different things. I could be wrong...

**Comment by getdave** [comment]
I figured this out. It's as @talldan suspected.

If you have no menus then inserting a Navigation block will create an "auto menu" which is basically a Page List block.

If you then use the Nav list view to insert a new block into this menu then a new Navigation menu (`wp_navigation`) will be created. However the creation of the menu causes focus to be lost from the list view and transfered to the editor iframe. 

Note how in the video below the focus gets transferred to the iframe when the menu is created. Note also that once the menu is created and I add a item to the menu, focus is handled correctly.

https://user-images.githubusercontent.com/444434/211777635-cf3b58ad-8350-4249-bb31-2e966a841095.mp4

When a menu is created the focus should remain in the Nav list view.

**Comment by getdave** [comment]
I tested this some more and I found that the selected block in the editor state is the Navigation block. But the focus is not on that block - somehow it ends up on an iframe.

I suspect this may be a wider bug...

**Comment by jasmussen** [comment]
CC: @ellatrix per the iframe.
